ocrd_utils/ocrd_utils/os.py

```diff
@@ -120,3 +120,44 @@ def get_fileobject(self, **kwargs):
 def atomic_write(fpath):
     with atomic_write_(fpath, writer_cls=AtomicWriterPerms, overwrite=True) as f:
         yield f
+
+def resolve_mets_arguments(directory, mets_url, mets_basename, log=None):
+    """
+    Resolve the ``--mets``, ``--mets-basename`` and `--directory`` argument
+    into a coherent set of arguments according to https://github.com/OCR-D/core/issues/517
+    """
+    if mets_basename and mets_url:
+        raise ValueError("Use either --mets or --mets-basename, not both")
+    elif not mets_basename and mets_url:
+        mets_basename = Path(mets_url).name
+    elif not mets_basename and not mets_url:
+        mets_basename = 'mets.xml'
+    else:
+        (log.warning if log else print)(DeprecationWarning("--mets-basename is deprecated. Use --mets/--directory instead"))
+
+    if directory and mets_url:
+        # XXX check whether mets_url has no parents, i.e. is actually the mets_basename
```

Comment:
Background: Due to how we set defaults in the click decorator, `--mets-url` will default to `mets.xml`. In that case (URL being a plain filename), behave as if `--mets-url` was omitted but `--mets-basename` was given. Not sure how robust that is though :grimacing:

Reply:
I'd prefer if the exception / check could remain in place. Why don't you simply do the `abspath` thing to the processor decorator's `--mets` option as well?

Reply:
Because `--mets` might be `mets.xml` (as is the case when calling a processor with `--working-dir` but without explicit `--mets`.

So for example: `ocrd-dummy --working-dir /path/to/workspace`:
* `abspath(--directory) == /path/to/workspace`
* `abspath(--mets)  == /home/current/working/dir/mets.xml`

Which would be inconsistent. This is the case where I think it makes sense to treat `--mets` as `--mets-basename`.

Reply:
Ah, I see, there **is** ambiguity in that CLI! So basically it does not make sense to provide `--mets` _and_ `--working-dir` – while it does to provide `--mets-basename` with `--working-dir` or `--mets` alone.

Couldn't we just make this implicit exclusiveness explicit by throwing an exception if both are used together (even if they do happen to contain the same directory prefix)? (Better simple and strict than complex and surprising...)

Reply:
> (Better simple and strict)

Agreed, however we do need the combination of --directory and --mets-url for the case that --mets-url is an HTTP(S) URL (think: `ocrd workspace clone`), so I've changed the resolution logic to be based on that, which hopefully makes things less ambiguous...